### DISCUSSION TITLE
long shot Q about JSON responses

### DISCUSSION THREAD

**Discussion by JeffMelton** [vmware/govmomi#3255] [Community Help] [answered]
I'm consuming a production vSphere 7.x API that I have no control over or insight into, and the JSON responses I get when using govc have changed the case used in the keys twice in the last couple weeks. That is, responses have been reliably PascalCase until the last couple of weeks. Then _some_ (but not all) of the keys switched to camelCase, with the remainder seeming still to be PascalCase. Then _more_ (but maybe not all?) of the keys are coming in camelCase. These are huge objects, so I can't be sure how consistent it is, but it's making me crazy.

I'm really just looking for a nudge in the right direction as to whether there's an update or a config flag somewhere that might have triggered those changes. Unless there's a govc config flag I can use to force that, this isn't govc-specific, but I figured I'd start here since that's what I'm using as my API client. TIA, and thanks for going on a tangent with me.

**Comment by dougm** [comment, marked as answer]
Hi @JeffMelton , sorry for the confusion. We did call out the breaking camelCase change in the [release notes](https://github.com/vmware/govmomi/releases/tag/v0.31.0).
The initial change was for only for generated types, to support the new [VI JSON API](https://developer.vmware.com/apis/vi-json/latest/)
However, the govc types were still PascalCase, which caused some confusion. We decided to also change the [govc types](https://github.com/vmware/govmomi/pull/3247) to camelCase, released a few days later and documented in the [release notes](https://github.com/vmware/govmomi/releases/tag/v0.32.0).

cc @akutz

**Reply by JeffMelton** [reply]
Aargh, that all checks out. I didn't even think about it being a govc issue. Thanks for clearing it up for me.
